core/src/main/java/io/questdb/cairo/AbstractIntervalDataFrameCursor.java

```diff
@@ -63,8 +63,15 @@ protected static long search(ReadOnlyColumn column, long value, long low, long h
                 low = mid + 1;
             else if (midVal > value)
                 high = mid;
-            else
+            else {
+                // In case of multiple equal values, find the first
+                int increment = scanUp ? -1 : 1;
+                for (long offset = 0; (mid + offset) > 0 && (mid + offset) < high; offset += increment) {
+                    if (midVal != column.getLong((mid + offset) * 8))
+                        return mid + offset - increment;
+                }
                 return mid;
```

Comment:
when loop goes all the way up with all values being the same the answer should be 0. Likewise if loop goes all the way down the answer should be end of column. `mid` is not a good outcome considering it is not mutated by the loop above.

Perhaps there is an opportunity to kill two birds with one stone and simplify loop by mutating `mid` ?

Reply:
Fixed.

Reply:
Perfect, thanks :+1: